src/jrd/tra.cpp

```diff
@@ -3579,6 +3585,15 @@ void jrd_tra::rollbackToSavepoint(thread_db* tdbb, SavNumber number)
  *
  **************************************/
 {
+#ifdef SAVEPOINT_DEBUG
+	printf("Request to rollback savepoint %" QUADFORMAT "d. Stack: ", number);
+	for (Savepoint* itr = tra_save_point; itr; itr = itr->getNext())
```

Comment:
On 05/26/2016 05:22 PM, Dimitry Sibiryakov wrote:
>> @@ -3579,6 +3585,15 @@ void jrd_tra::rollbackToSavepoint(thread_db* tdbb, SavNumber number)
>>    *
>>    **************************************/
>>   {
>> +#ifdef SAVEPOINT_DEBUG
>> +	printf("Request to rollback savepoint %" QUADFORMAT "d. Stack: ", number);
>> +	for (Savepoint* itr = tra_save_point; itr; itr = itr->getNext())
> Ok.

And it should go to stderr, not stdout.
Only stderr (2) remains opened on posix when  -d switch is used, stdout 
(1) can become network socket or something else.